Scripts/RemediationScripts/Remediate-EnableEncryptionInTransitForStorageAccounts.ps1

```diff
@@ -323,147 +413,269 @@ function Enable-StorageEncryptionInTransit
+                $logResource = @{}
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+                $logResource.Add("ResourceName",($_.ResourceName))
+                $logResource.Add("Reason","Secure Transfer is already enabled.")    
+                $logSkippedResources += $logResource
             }        
             else
             { 
-                 $stgWithDisableHTTPS += $_|Select-Object  @{N='ResourceId';E={$_.Id}},
-                                                           @{N='ResourceGroupName';E={$_.ResourceGroupName}},
-                                                           @{N='StorageAccountName';E={$_.StorageAccountName}},
-                                                           @{N='EnableHttpsTrafficOnly';E={$_.EnableHttpsTrafficOnly}}                                                                  
+                $stgWithDisableHTTPS += $_|Select-Object  @{N='ResourceId';E={$_.Id}},
+                                                        @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                        @{N='StorageAccountName';E={$_.StorageAccountName}},
+                                                        @{N='EnableHttpsTrafficOnly';E={$_.EnableHttpsTrafficOnly}}                                                                  
             }
-        }                    
-                              
-        $totalStgWithEnableHTTPS = ($stgWithEnableHTTPS | Measure-Object).Count
-        $totalStgWithDisableHTTPS = ($stgWithDisableHTTPS | Measure-Object).Count
-    
-        Write-Host "Storage Account(s) with enabled 'secure transfer': [$($totalStgWithEnableHTTPS)]" -ForegroundColor $([Constants]::MessageType.Update)
-        Write-Host "Storage Account(s) with disabled 'secure transfer': [$($totalStgWithDisableHTTPS)]" -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host "Successfully fetched secure tranfer configuration of Storage Account: [$($_.StorageAccountName)]" -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+        catch {
+            Write-Host "Error encountered while fetching secure transfer configuration of the Storage Account: [$($_.StorageAccountName)]." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host "Skipping this Storage Account..." -ForegroundColor $([Constants]::MessageType.Warning)
+            $logResource = @{}
+            $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+            $logResource.Add("ResourceName",($_.ResourceName))
+            $logResource.Add("Reason","Error encountered while fetching secure transfer configuration of the Storage Account.")    
+            $logSkippedResources += $logResource
+            Write-Host $([Constants]::SingleDashLine)
+        }
+    }  
+    $totalStgWithEnableHTTPS = ($stgWithEnableHTTPS | Measure-Object).Count
```

Comment:
Summary count of HTTPS enabled, disabled, and errored storage account will help here.

Reply:
Is errored one necessary?

Reply:
Write-Host "Storage Accounts(s) where error was encountered while fetching configuration details: [$($totalStgEncounteredError)]" -ForegroundColor $([Constants]::MessageType.Update)


good enough?

Reply:
Yeah